app/services/UIBuilder.scala

```diff
@@ -138,4 +139,42 @@ class UIBuilder {
       case x :: xs => partitionComponents(xs, errors, x +: others)
     }
 
+  private def fromSequenceWithLeadingCallout(sg: StackedGroup, calloutType: CalloutType, formData: Option[FormData])
```

Comment:
would be simpler to make this YourCall specific. Callout in fn name should be Callouts as we know we have at least stacked callouts

Reply:
Originally it was a specific function for YourCall callouts, but I think there will be a case for other callout types to be handled in the same manner. As such I think this approach will be better in the long run. However, you are right about the name.

Reply:
Following  the removal of "extractLeadingCallouts", see below, I think it would be better to make the function specific to call outs of type yourCall.